src/pages/InsightPage/InsightPage.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { useState } from "react";
+import { DragDropContext, Draggable, DropResult, Droppable } from 'react-beautiful-dnd';
+import { ContentCard } from '../../components/Cards/ContentCard';
+import { InsightRow } from "../../components/InsightRow";
+import './style.css';
+
+const InsightPage = () => {
+    const insightRows = [
+        { id:1, title: "Improve Service Level", color: "white" },
+        { id:2, title: "Improve Occupancy", color: "white" },
+        { id:3, title: "Reassignment", color: "white" },
+    ];
+    const kanbanInsight = {
+        'To-do': {
+            id: 'To-do',
+            list: insightRows
+        },
+        'In Progress': {
+            id: 'In Progress',
+            list: []
+        },
+        'Done': {
+            id: 'Done',
+            list: []
+        }
+    }
+
+    const [kanban, setKanban] = useState(kanbanInsight)
+
+    const onDragEnd = ({source,destination}: DropResult) => {
+        if (!destination) return;
+    
+        if (source.droppableId === destination.droppableId && source.index === destination.index) {
+            return;
+        }
+    
+        const start = kanban[source.droppableId as keyof typeof kanban];
+        const end = kanban[destination.droppableId as keyof typeof kanban];
+        if (start === end) {
+            const newList = Array.from(start.list);
+            const [draggableItem] = newList.splice(source.index, 1);
+            newList.splice(destination.index, 0, draggableItem);
+    
+            setKanban(prevState => ({
+                ...prevState,
+                [start.id]: {
+                    ...start,
+                    list: newList
+                }
+            }));
+        } else { 
+            const startList = Array.from(start.list);
+            const [draggableItem] = startList.splice(source.index, 1);
+            const endList = Array.from(end.list);
+            endList.splice(destination.index, 0, draggableItem);
+    
+            setKanban(prevState => ({
+                ...prevState,
+                [start.id]: {
+                    ...start,
+                    list: startList
+                },
+                [end.id]: {
+                    ...end,
+                    list: endList
+                }
+            }));
+        }
+    };
+    
+    
+    return (
+    <DragDropContext onDragEnd={onDragEnd}>
```

Comment:
CÓMO QUE HICISTE EL DRAG AND DROPPPPPPP SOY TU FANNNNNN MUCHAS GRACIAS DAME TU AUTÓGRAFO PUÑIS